x/programs/rust/examples/counter/src/lib.rs

```diff
@@ -6,27 +6,6 @@ pub enum StateKeys {
     Counter(Address),
 }
 
-/// Initializes the program address a count of 0.
-#[public]
-pub fn initialize_address(context: Context<StateKeys>, address: Address) -> bool {
-    let Context { program, .. } = context;
-
-    if program
-        .state()
-        .get::<i64>(StateKeys::Counter(address))
-        .is_ok()
-    {
-        panic!("counter already initialized for address")
-    }
-
-    program
-        .state()
-        .store(StateKeys::Counter(address), &0_i64)
-        .expect("failed to store counter");
-
-    true
-}
```

Comment:
Fallible initialize never really made sense. If we wanted a counter with explicit initialization, we should have been able to set the number, otherwise it's wasteful to store the 0 case.

Reply:
Totally agree, otherwise we could have had a function called `initialized` that returns

```rust
    if program
        .state()
        .get::<i64>(StateKeys::Counter(address))
        .is_ok()
  ```

that would be called after the first call to `inc*` in tests, but that is effectively the same sanity check as the next calls to `get` so I'm fine just removing that function